From 84e5868d9ca60b13b9156cd9abdf4b67420ffb2e Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Tue, 26 Nov 2024 09:49:22 +0100
Subject: [PATCH 1369/1695] drivers: crypto: remove 192bits key software
 fallback for stm32mp21

SAES peripheral has been updated on STM32MP21 and now support 192 bits
keys. Hence, remove software fallback used for STM32MP23/25 to support
192 bits keys.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fed875e10243dcc2a7561bcda4b86ec7cc89f32e drivers: crypto: remove 192bits key software fallback for stm32mp21]
Change-Id: Idd6810d935fc13970083eeceb8384505bc04270a
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420725
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/crypto/stm32/cipher.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/crypto/stm32/cipher.c b/core/drivers/crypto/stm32/cipher.c
index 95410ac16..d1524ba72 100644
--- a/core/drivers/crypto/stm32/cipher.c
+++ b/core/drivers/crypto/stm32/cipher.c
@@ -112,7 +112,7 @@ static TEE_Result saes_init(union ip_ctx *ip_ctx, bool is_decrypt,
 	if (!IS_ENABLED(CFG_STM32_SAES))
 		return TEE_ERROR_NOT_IMPLEMENTED;
 
-	if (key_len == AES_KEYSIZE_192) {
+	if (!IS_ENABLED(CFG_STM32MP21) && key_len == AES_KEYSIZE_192) {
 		struct crypto_cipher_ctx *ctx = ip_ctx->saes.fallback_ctx;
 		TEE_OperationMode mode = TEE_MODE_ILLEGAL_VALUE;
 		TEE_Result res = TEE_ERROR_GENERIC;
-- 
2.47.2

